src/entities/User.ts

```diff
@@ -1,9 +1,12 @@
 import {
+  PrimaryGeneratedColumn,
   Column,
   Entity,
-  PrimaryGeneratedColumn,
+  ManyToMany,
+  JoinTable
 } from 'typeorm';
-import { SerializedUser } from '../common/types';
+import { SerializedUser, SubSerializedUser } from '../common/types';
```

Comment:
what exactly is a subserializeduser

Reply:
do you have a comment anywhere that describes it cuz i is a confuseboi

Reply:
A subserializeduser represents a User without their matches shown. It's used so that we can show a user's matchings without causing an infinite loop where calling matching.serialize() calls user.serialize() calls matching.serialize() ... 

I added comments to explain all types in the types.ts file